src/content/docs/tutorials/rust/rust_tutorial.mdx

```diff
@@ -7,181 +7,211 @@ import { Tabs, TabItem } from '@astrojs/starlight/components';
-as per our desired schema as follows:
+Kuzu also supports an `in-memory mode` where the database is active only during the execution of the Rust program. For more information, refer to the example in the [create your first graph](/get-started/#index) section.
+
+### Define schema
+The first step in building a Kuzu graph is to define a schema. For our example dataset, we need the following node and relationships tables:
 ```rs
-conn.query("
-CREATE NODE TABLE IF NOT EXISTS User(
-    user_id INT64 PRIMARY KEY,
-    username STRING,
-    account_creation_date DATE
-)
-")?;
-conn.query("
+conn.query(
+    "
+    CREATE NODE TABLE IF NOT EXISTS User(
+        user_id INT64 PRIMARY KEY,
+        username STRING,
+        account_creation_date DATE
+    )",
+)?;
+conn.query(
+    "
     CREATE NODE TABLE IF NOT EXISTS Post(
-        post_id INT64 PRIMARY KEY, 
-        post_date DATE, 
-        like_count INT64, 
+        post_id INT64 PRIMARY KEY,
+        post_date DATE,
+        like_count INT64,
         retweet_count INT64
-    )")?;
-conn.query("
+    )",
+)?;
+conn.query(
+    "
     CREATE REL TABLE IF NOT EXISTS FOLLOWS(
         FROM User TO User
-    )")?;
-conn.query("
+    )",
+)?;
+conn.query(
+    "
     CREATE REL TABLE IF NOT EXISTS POSTS(
         FROM User TO Post
-    )")?;
-conn.query("
+    )",
+)?;
+conn.query(
+    "
     CREATE REL TABLE IF NOT EXISTS LIKES(
         FROM User TO Post
-    )")?;
+    )",
+)?;
 ```
 
-## Show table information
+### Import CSV data
 
-We can check information on the tables we have in the database with the following commands:
+We can now import the CSV data into the node and relationship tables:
+```rs
+conn.query("COPY User FROM './tutorial_data/node/user.csv'")?;
+conn.query("COPY Post FROM './tutorial_data/node/post.csv'")?;
+conn.query("COPY FOLLOWS FROM './tutorial_data/relation/FOLLOWS.csv'")?;
+conn.query("COPY POSTS FROM './tutorial_data/relation/POSTS.csv'")?;
+conn.query("COPY LIKES FROM './tutorial_data/relation/LIKES.csv'")?;
+```
+See the documentation on [importing data](/import/csv) for more details on the parameters for CSV import.
+
+### Show table information
+
+Finally, we can check the schema of the newly created tables:
 ```rs
 let mut result = conn.query("CALL SHOW_TABLES() RETURN *")?;
 println!("{}", result.display());
 ```
-Returns:
-```
+### Running the code
+
+Run `create-db.rs` to create the Kuzu database and import the CSV files:
+```bash
+$ cargo run --bin create-db
 id|name|type|database name|comment
 0|User|NODE|local(kuzu)|
-1|Post|NODE|local(kuzu)|
 2|FOLLOWS|REL|local(kuzu)|
+1|Post|NODE|local(kuzu)|
 3|POSTS|REL|local(kuzu)|
 4|LIKES|REL|local(kuzu)|
 ```
+The CSV data has been copied into a new Kuzu database in `./social_network_db`.
+
+:::note[Note]
+If you run the program again, you'll get an error about duplicated primary keys.
+This is because the data has already been imported into the Kuzu database.
```

Comment:
If there's going to be an error anyway if you run it multiple times, do we really need to use "IF NOT EXISTS" when creating the schema?